engines/lure/fights.cpp

```diff
@@ -190,26 +190,25 @@ void FightsManager::checkEvents() {
 	int moveNumber = 0;
 
 	while ((moveNumber == 0) && events.pollEvent()) {
-
-		if (events.type() == Common::EVENT_KEYDOWN) {
-			switch (events.event().kbd.keycode) {
-			case Common::KEYCODE_ESCAPE:
+		if (events.type() == Common::EVENT_CUSTOM_ENGINE_ACTION_START) {
+			switch (events.event().customType) {
+			case kActionEscape:
 				engine.quitGame();
 				return;
-
 			default:
 				// Scan through the mapping list for a move for the keypress
-				const KeyMapping *keyPtr = &keyList[0];
-				while ((keyPtr->keycode != Common::KEYCODE_INVALID) &&
-					(keyPtr->keycode != events.event().kbd.keycode))
-					++keyPtr;
-				if (keyPtr->keycode != Common::KEYCODE_INVALID) {
-					moveNumber = keyPtr->moveNumber;
+				const ActionMapping *actionPtr = &actionList[0];
+				while ((actionPtr->customAction != kActionNone) &&
+					   (actionPtr->customAction != events.event().customType))
+					++actionPtr;
+				if (actionPtr->customAction != kActionNone) {
+					moveNumber = actionPtr->moveNumber;
 					_keyDown = KS_KEYDOWN_1;
 				}
+
 			}
 
-		} else if (events.type() == Common::EVENT_KEYUP) {
+		} else if (events.customType() == Common::EVENT_CUSTOM_ENGINE_ACTION_END) {
```

Comment:
I believe this was a mistake - the event type is not same as the event custom type, correct?

Reply:
Fixed.